.github/workflows/tests.yaml

```diff
@@ -0,0 +1,182 @@
+# Run all tests, with or without flaky tests.
+
+name: Tests
+
+on:
+  workflow_call:
+    inputs:
+      flaky:
+        description: 'Whether to also run flaky tests'
+        type: boolean
+        default: false
+      git-ref:
+        description: 'Which git ref to checkout'
+        type: string
+        default: 'main'
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  RUST_BACKTRACE: 1
+  RUSTFLAGS: -Dwarnings
+  RUSTDOCFLAGS: -Dwarnings
+  SCCACHE_CACHE_SIZE: "50G"
+
+jobs:
+  build_and_test_nix:
+    timeout-minutes: 30
+    name: "Tests"
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [ubuntu-latest, macOS-arm-latest]
+        rust: [nightly, stable]
+        features: [all, none, default]
+        include:
+          - name: ubuntu-latest
+            os: ubuntu-latest
+            release-os: linux
+            release-arch: amd64
+            runner: [self-hosted, linux, X64]
+          - name: macOS-arm-latest
+            os: macOS-latest
+            release-os: darwin
+            release-arch: aarch64
+            runner: [self-hosted, macOS, ARM64]
+    env:
+      # Using self-hosted runners so use local cache for sccache and
+      # not SCCACHE_GHA_ENABLED.
+      RUSTC_WRAPPER: "sccache"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.git-ref }}
+
+    - name: Install ${{ matrix.rust }}
+      uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: ${{ matrix.rust }}
+
+    - name: Install cargo-nextest
+      uses: taiki-e/install-action@v2
+      with:
+        tool: nextest
+    
+    - name: Install sccache
+      uses: mozilla-actions/sccache-action@v0.0.3
+
+    - name: Select features
+      run: |
+        case "${{ matrix.features }}" in
+            all)
+                echo "FEATURES=--all-features" >> "$GITHUB_ENV"
+                ;;
+            none)
+                echo "FEATURES=--no-default-features" >> "$GITHUB_ENV"
+                ;;
+            default)
+                echo "FEATURES=" >> "$GITHUB_ENV"
+                ;;
+            *)
+                exit 1
+        esac
+
+    - name: check features
+      if: ${{ ! inputs.flaky }}
```

Comment:
is `inputs.xyz` the correct syntax here. Afaik these should be subclassed to `github.event.inputs` or `github.inputs`

Reply:
i believe `inputs` is correct: https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context

Reply:
Nice, I used to think it was always scoped to the `github` context.